src/main/java/org/opensrp/queue/QueueHelper.java

```diff
@@ -0,0 +1,33 @@
+package org.opensrp.queue;
+
+import org.opensrp.service.PlanService;
+import org.smartregister.domain.Jurisdiction;
+import org.smartregister.domain.PlanDefinition;
+import org.smartregister.pathevaluator.TriggerType;
+import org.smartregister.pathevaluator.dao.QueuingHelper;
+import org.springframework.stereotype.Component;
+
+
+@Component
+public class QueueHelper implements QueuingHelper {
+
+	PlanService planService;
+
+	RabbitMQSender rabbitMQSender;
+
+	@Override
+	public void addToQueue(String planIdentifier, TriggerType triggerType, String locationId) {
+		PlanDefinition planDefinition = planService.getPlan(planIdentifier);
```

Comment:
Could we send just the plan definition Id, we can query for the plan when receiving/processing the message

Reply:
Otherwise pass over the plan definition directly from plan evaluator library

Reply:
@rehammuzzamil pass the the plan Identifier and query the plan on the receiving job